ghtt/assignment.py

```diff
@@ -109,14 +109,18 @@ def generate_file_from_template(path, clone_url, repo: ghtt.config.StudentRepo):
     `.jinja`, the template file is removed and the result is saved without that extension. If not,
     the template file is overwritten with the generated result.
     """
-    template = open(path).read()
-    outputText = render_template(template, clone_url, repo)
-    destination = str(path)
-    if destination.endswith('.jinja'):
-        os.remove(path)
-        destination = destination[:-6]
-    with open(destination, "w+") as d_file:
-        d_file.write(outputText)
+    print("TEMPLATE: ", open(path).read())
+    try:
+        template = open(path).read()
+        outputText = render_template(template, clone_url, repo)
+        destination = str(path)
+        if destination.endswith('.jinja'):
+            os.remove(path)
+            destination = destination[:-6]
+        with open(destination, "w+") as d_file:
+            d_file.write(outputText)
+    except:
```

Comment:
In general, it's recommended not to use a blanked `except` without specifying the specific exceptions that you're catching. This makes sure that when an "unexpected" exception happens, it's still surfaced and not silently hidden.

In this case, for example, it might be interesting to know what failed; did render_template fail, and if so why? Was the file not found? Was there are permissions error? Because of this reason, I'm also not a big fan of catching exceptions just to raise new ones.

What is the idea behind this change? Maybe we can find a different way to achieve the functionality you want?

Reply:
Yes, PEP8 says you shouldn't use base `except`. That's good advice in the general case, but there are actually plenty of good cases where it's OK to do it anyway imho.

In this specific case, the idea is to add more context to the error:
```
  except:
    raise BaseException('Problem generating template for path={} clone_url={}'.format(path, clone_url))
```
adds `path` and `clone_url`, which would otherwise be lost and not available in the output/logs.

Note that raising an exception inside `except` does not hide the original exception as far as I know. The traceback of both exceptions will be logged in my experience (with messages like "during the handling of the above exception, the below exception occured"). So no information is lost, we just added some.

But I agree that you could replace this case with this cleaner version:
```
try:
   ...
except:
   logger.error(f'Exception while generating template for path={path} clone_url={clone_url}')
   raise
```

Reply:
Ok, the cleaner version looks good and explicitly shows the original exception will be raised.